mart/attack/adversary.py

```diff
@@ -299,7 +305,7 @@ def forward(
         target: torch.Tensor | dict[str, Any] | tuple,
         **kwargs,
     ):
-        perturbation = self.perturber(input, target)
+        perturbation = self.perturber(input=input, target=target)
         output = self.composer(perturbation, input=input, target=target)
 
         return output
```

Comment:
I still don't know why an `Adversary` needs to know about the perturber and composer. Shouldn't those modules just be in the SequentialDict list of modules?

Reply:
I guess it does need to know the set of parameters to optimize. How can that be easily done?

Reply:
My sense is that we need to merge Adversary and Perturber.